Pipeline/CreateUCDComponentVersion/README.md

```diff
@@ -1,69 +1,239 @@
-# IBM UCD Packaging based on IBM DBB Build Report
+# IBM UCD Packaging Based on IBM DBB Build Report
 
-An important step in the pipeline is to generate a deployable package. This sample groovy script
-- extracts information about the build outputs from the DBB BuildReport.json
-- generates the UCD shiplist ```shiplist.xml``` file
-- invokes the ```buztool.sh``` of UCD
+## Summary
+
+An important step in the pipeline is to generate a deployable package. This sample Groovy script:
+
+- Extracts information about the build outputs from the Dependency Based Build (DBB) `BuildReport.json`. The script is able to take a single DBB build report or multiple build reports to build a cumulative package across multiple incremental builds. 
+- Generates the UrbanCode Deploy (UCD) shiplist `shiplist.xml` file.
+- Invokes the `buztool.sh` with the appropriate configuration to store the binary package in the artifact repository and to register a new UCD component version.
+
+## High-level Processing Flow
+
+1. **Initialization**
+   1. Read command line parameters.
+   1. (Optional) Read any application and global properties that are passed via `--packagingPropFiles`.
+
+1. **Process the DBB build report(s)**
+   1. Either read DBB's `BuildReport.json` from the pipeline work directory, or loop through the list of provided DBB build reports (using the `--buildReportOrder` or `--buildReportOrderFile` option).
+   1. Parse and extract build output information for records of type *ExecuteRecord* and *CopyToPDSRecord*. (Requires at least DBB 1.0.8.)
+   1. Parse and extract the build output information for deleted build outputs of type *Delete_Record* written to the build report by zAppBuild. (Requires at least DBB 1.1.3, as the script uses the AnyTypeRecord API introduced in this version.)
+
+1. **Generate the UCD `shiplist.xml` file and invoke UCD packaging step**
+   1. Write `shiplist.xml` to the build directory:
+      1. (Optional) Add links back to the continuous integration (CI) pipeline build, the Git pull request to UCD component version.
+      1. (Optional) Add UCD artifact level properties for bind information captured in the zAppBuild framework through generic PropertyRecords for DBRM members, such as `bind_collectionID`, `bind_packageOwner`, `bind_qualifier` on the element level - see [generateDb2BindInfoRecord configuration in zAppBuild](https://github.com/IBM/dbb-zappbuild/blob/06ff114ee22b4e41a09aa0640ac75b7e56c70521/build-conf/build.properties#L79-L89).  
```

Comment:
Just to clarify, is this step referring to adding properties at the artifact level (in which case I would say "artifact-level properties")? Or is there something called an "artifact level" to which the script is adding properties?